File: packages/ruby/lib/tesote_sdk/v2/sync_sessions.rb
module TesoteSdk
  module V2
    class SyncSessions
      def initialize(transport)
        @transport = transport
      end

      # GET /v2/accounts/{id}/sync_sessions
      def list(account_id, query = {}, opts: {})
        raise ArgumentError, 'account_id is required' if account_id.nil? || account_id.to_s.empty?

        body = @transport.request('GET', "accounts/#{account_id}/sync_sessions", query: query, opts: opts)
        items_raw = (body && body['sync_sessions']) || []
        Models::OffsetPage.new(
          items: Models::SyncSession.from_array(items_raw),
          limit: body && body['limit'],
          offset: body && body['offset'],
          has_more: body && body['has_more']
        )
      end

      # Walks pages via offset pagination; yields OffsetPage per call.
      def each_page(account_id, query = {}, opts: {}, page_size: 50, &block)
        return enum_for(:each_page, account_id, query, opts: opts, page_size: page_size) unless block

        enum = Pagination::OffsetEnumerator.new(start_query: query, limit: page_size) do |q|
          list(account_id, q, opts: opts)
        end
        enum.each(&block)
      end

      # GET /v2/accounts/{id}/sync_sessions/{session_id}
      def get(account_id, session_id, opts: {})
        raise ArgumentError, 'account_id is required' if account_id.nil? || account_id.to_s.empty?
        raise ArgumentError, 'session_id is required' if session_id.nil? || session_id.to_s.empty?

        body = @transport.request('GET', "accounts/#{account_id}/sync_sessions/#{session_id}", opts: opts)
        Models::SyncSession.from_hash(body)
      end
    end
  end
end

File: packages/python/src/tesote_sdk/v2/sync_sessions.py
"""v2 sync_sessions resource: list + get per account."""

from __future__ import annotations

from typing import Any, Dict, Iterator, Optional

from ..models import SyncSession, SyncSessionList
from ..transport import Transport


class SyncSessionsResource:
    """`/v2/accounts/{id}/sync_sessions` -- list + show."""

    def __init__(self, transport: Transport) -> None:
        self._transport = transport

    def list(
        self,
        account_id: str,
        *,
        limit: Optional[int] = None,
        offset: Optional[int] = None,
        status: Optional[str] = None,
        cache_ttl: Optional[float] = None,
    ) -> SyncSessionList:
        query: Dict[str, Any] = {
            "limit": limit,
            "offset": offset,
            "status": status,
        }
        response = self._transport.request(
            "GET",
            f"/v2/accounts/{account_id}/sync_sessions",
            query=query,
            cache_ttl=cache_ttl,
        )
        body: Dict[str, Any] = response.json if isinstance(response.json, dict) else {}
        return SyncSessionList.from_dict(body)

    def iter(
        self,
        account_id: str,
        *,
        status: Optional[str] = None,
        page_size: int = 50,
    ) -> Iterator[SyncSession]:
        """Iterate every sync session for the account using offset pagination."""
        offset = 0
        while True:
            page = self.list(
                account_id, limit=page_size, offset=offset, status=status
            )
            if not page.sync_sessions:
                return
            yield from page.sync_sessions
            if not page.has_more:
                return
            offset += len(page.sync_sessions)

    def get(
        self,
        account_id: str,
        session_id: str,
        *,
        cache_ttl: Optional[float] = None,
    ) -> SyncSession:
        response = self._transport.request(
            "GET",
            f"/v2/accounts/{account_id}/sync_sessions/{session_id}",
            cache_ttl=cache_ttl,
        )
        body: Dict[str, Any] = response.json if isinstance(response.json, dict) else {}
        return SyncSession.from_dict(body)


__all__ = ["SyncSessionsResource"]
